packages/renderers/renderer-solid/client.js

```diff
@@ -1,15 +1,16 @@
-import { createComponent } from 'solid-js/web';
+import { createComponent } from 'solid-js';
+import { render } from 'solid-js/web';
 
-export default (element) => (Component, props) => {
-  // Solid `createComponent` just returns a DOM node with all reactivity
-  // already attached. There's no VDOM, so there's no real need to "mount".
-  // Likewise, `children` can just reuse the nearest `astro-fragment` node.
-  const component = createComponent(Component, {
-    ...props,
-    children: element.querySelector('astro-fragment'),
-  });
+export default (element) => (Component, props, childHTML) => {
+  // Solid's `render` does not replace the element's children.
+  // Deleting the root's children is necessary before calling `render`.
+  element.replaceChildren();
```

Comment:
Won't this blow away state such as focused elements? Does Solid not support hydrating existing DOM?

Reply:
Should we be using this instead? https://www.solidjs.com/docs/1.0.0#hydrate

Reply:
Hey, @matthewp, thanks a lot for your time! This is similar to the previous implementation which was also destructive. Just using the `render` method to mount the root.

Solid _does_ support hydration but as far as I can tell it needs a script to be [generated and added to the head in the server](https://www.solidjs.com/guide#hydration-script). It does not throw any error if I generate it per component on `server.js` (but that'd be adding the same `<script>` tag over and over again per component. I see you can run a script before hydration using `hydrationPolyfills` or `polyfills` in `index.js` but in this case I'm not sure how important it is for this script tag to be _generated_.

A more important issue with hydration is that components that have children passed by Astro do not get hydrated at all. I'm guessing it has to do with [Solid not having generated those components](https://www.solidjs.com/guide#ssr-caveats). I've been testing locally and using `render` if children are passed and `hydrate` when children are not passed seems to work. But the most consistent solution seems to be to completely replace it with `render`.

Reply:
Scratch that last part, hydration failing like that seems to happen only when the top component is a fragment with more than one child element.

On another note, though. Using `hydrate` on a component that has a `Dynamic` somewhere within it fails completely.

These both seem to be issues within Solid, though. I'll mention them over there.